File: c++/10253.cpp
/*
    10253 : 헨리
    URL : https://www.acmicpc.net/problem/10253
    Input :
        3
        4 23
        5 7
        8 11
    Output :
        138
        70
        4070
*/
#include <iostream>

using namespace std;

int find_henry(int p, int q);

int main()
{
    int T;
    cin >> T;
    while (T--)
    {
        int a, b;
        cin >> a >> b;
        cout << find_henry(a, b) << endl;
    }
    return 0;
}

int find_henry(int p, int q)
{
    while(p > 1)
    {
        if (q % p == 0)
        {
            q = q / p;
            p = p / p;
            if (p == 1)
            {
                return q;
            }
        }
        int x = (q / p) + 1;
        p = p * x - q;
        q = q * x;
    }
    return q;
}

File: baekjoon/10253.py
"""
    10253 : 헨리
    URL : https://www.acmicpc.net/problem/10253
    Input :
        3
        4 23
        5 7
        8 11
    Output :
        138
        70
        4070
"""


def gcd(a, b):
    if b > a:
        return gcd(b, a)

    if b == 0:
        return a

    if (a % b) == 0:
        return b
    else:
        return gcd(b, a % b)


def smaller(a, b, x):
    return (a * x - b) >= 0


def minus(a, b, x):
    return (a * x - b), (b * x)


def henry(a, b):
    g = gcd(a, b)
    if g != 1:
        a = a // g
        b = b // g

    if a == 1:
        return b

    x = 2
    while True:
        if smaller(a, b, x):
            break
        x += 1

    a, b = minus(a, b, x)
    return henry(a, b)


T = int(input())
for t in range(T):
    a, b = map(int, input().split())
    print(henry(a, b))
